lib/mayaUsd/fileio/orphanedNodesManager.cpp

```diff
@@ -110,14 +112,31 @@ void OrphanedNodesManager::operator()(const Ufe::Notification& n)
             }
         }
     } else if (pulledPrims().containsDescendant(changedPath)) {
+#ifdef UFE_V4_FEATURES_AVAILABLE
+        // Use UFE v4 notification to op conversion.
         handleOp(sceneNotification);
+#else
+        // UFE v3: convert to op ourselves.  Only convert supported
+        // notifications.
+        if (auto objAdd = dynamic_cast<const Ufe::ObjectAdd*>(&sceneNotification)) {
+            handleOp(Ufe::SceneCompositeNotification::Op(
+                Ufe::SceneCompositeNotification::ObjectAdd, objAdd->item()));
+        } else if (auto objDel = dynamic_cast<const Ufe::ObjectDelete*>(&sceneNotification)) {
+            handleOp(Ufe::SceneCompositeNotification::Op(
+                Ufe::SceneCompositeNotification::ObjectDelete, objDel->path()));
+        } else if (
+            auto subtrInv = dynamic_cast<const Ufe::SubtreeInvalidate*>(&sceneNotification)) {
+            handleOp(Ufe::SceneCompositeNotification::Op(
+                Ufe::SceneCompositeNotification::SubtreeInvalidate, subtrInv->root()));
+        }
+#endif
     }
 }
 
 void OrphanedNodesManager::handleOp(const Ufe::SceneCompositeNotification::Op& op)
 {
     switch (op.opType) {
-    case Ufe::SceneChanged::ObjectAdd: {
+    case Ufe::SceneCompositeNotification::ObjectAdd: {
```

Comment:
In UFE v3 the enumerant was in the SceneCompositeNotification derived class, in UFE v4 it's move to the base class, but the derived class scope still works, so using the derived class is UFE v3/v4 compatible, a requirement for 2023 compatibility.